src/components/views/rooms/EventTile.tsx

```diff
@@ -923,6 +938,74 @@ export class UnwrappedEventTile extends React.Component<IProps, IState> {
         });
     };
 
+    private renderContextMenu(): React.ReactFragment {
+        let contextMenu = null;
+        if (this.state.contextMenu) {
+            const tile = this.getTile();
+            const replyChain = this.getReplyChain();
+            const eventTileOps = tile?.getEventTileOps ? tile.getEventTileOps() : undefined;
+            const collapseReplyChain = replyChain?.canCollapse() ? replyChain.collapse : undefined;
+
+            contextMenu = (
+                <MessageContextMenu
+                    {...aboveLeftOf(this.state.contextMenu.position)}
+                    mxEvent={this.props.mxEvent}
+                    permalinkCreator={this.props.permalinkCreator}
+                    eventTileOps={eventTileOps}
+                    collapseReplyChain={collapseReplyChain}
+                    onFinished={this.onCloseMenu}
+                    rightClick={true}
+                    reactions={this.state.reactions}
+                    showPermalink={this.state.contextMenu.showPermalink}
+                />
+            );
+        }
+
+        return (
+            <React.Fragment>
+                { contextMenu }
+            </React.Fragment>
+        );
```

Comment:
ideally we return `null` when there's no menu rather than an empty fragment, for a slight performance boost.